src/redisearch_rs/inverted_index_bencher/src/ffi.rs

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+use std::{
+    alloc::{Layout, alloc},
+    ffi::c_char,
+    ptr::NonNull,
+};
+
+use buffer::{Buffer, BufferReader, BufferWriter};
+
+mod bindings {
+    #![allow(non_snake_case)]
+    #![allow(non_upper_case_globals)]
+    #![allow(non_camel_case_types)]
+    #![allow(unsafe_op_in_unsafe_fn)]
+    #![allow(improper_ctypes)]
+    #![allow(dead_code)]
+
+    use inverted_index::{RSIndexResult, RSOffsetVector, t_docId, t_fieldMask};
+
+    include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
+}
+
+/// An extra wrapper around the ['Buffer`] which allows the C code to grow it correctly and for its
+/// memory to be cleaned up when it goes out of scope.
+pub struct TestBuffer(Buffer);
+
+impl TestBuffer {
+    pub fn new() -> Self {
+        let layout = Layout::array::<u8>(0).unwrap();
+        let data = unsafe { alloc(layout) };
```

Comment:
This  will break when we use `buffer.c` functions, like `Buffer_Grow`.
I think we need to use `allocator.rs` here instead, or `Buffer_Init` ?